static/js/composebox_typeahead.js

```diff
@@ -88,6 +88,35 @@ var nextFocus = false;
 function handle_keydown(e) {
     var code = e.keyCode || e.which;
 
+    if ((code === 66 || code === 73 || code === 75) && (e.ctrlKey || e.metaKey)) {
+        var textarea = $("#new_message_content");
+        var range = textarea.range();
+        function add_markdown(markdown) {
+        if (!document.execCommand('insertText', false, markdown)) {
+            textarea.range(range.start, range.end).range(markdown);
+            }
+        }
+
+        if (range.length > 0) {
+            if (code === 66) {
+                // convert selected text to bold text
+                add_markdown("**"+range.text+"**");
+            }
+            if (code === 73) {
+                // convert selected text to italic text
+                add_markdown("*"+range.text+"*");
+            }
+            if (code === 75) {
+                // insert a link to selected text
+                add_markdown("["+range.text+"]()");
```

Comment:
For this one, I think we might want slightly more complicated logic that leaves the cursor in between the two parens.

Reply:
Might need to add the **url** text between the parens here to make it more clear.
```javascript
add_markdown("["+range.text+"](url)");
```

Reply:
For cursor logic -
<img src="https://d26dzxoao6i3hh.cloudfront.net/items/3L2h1q2x0P1l3l1P2L0k/Screen%20Recording%202017-11-24%20at%2003.58%20PM.gif?v=b2e11fb7">